src/test/kotlin/com/terraformation/backend/customer/AppNotificationServiceTest.kt

```diff
@@ -119,7 +121,7 @@ internal class AppNotificationServiceTest : DatabaseTest(), RunsAsUser {
             organizationStore,
             ParentStore(dslContext),
             PermissionStore(dslContext),
-            mockk(),
+            publisher,
```

Comment:
use the publisher to publish events in the tests below - instead of current approach `service.on(event)` (sadly my doing)

Reply:
The approach you used is the right one. This dummy "publisher" doesn't call any event listeners, and the listeners are what we're testing.